src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -442,23 +453,39 @@ public void run() throws InterruptedException {
                 // operations. This ensures the shard engine is started and disables
                 // garbage collection (not the JVM's GC!) of tombstone deletes
                 transportService.submitRequest(request.targetNode(), RecoveryTarget.Actions.PREPARE_TRANSLOG,
-                        new RecoveryPrepareForTranslogOperationsRequest(request.recoveryId(), request.shardId(), shard.translog().estimatedNumberOfOperations()),
+                        new RecoveryPrepareForTranslogOperationsRequest(request.recoveryId(), request.shardId(), translogView.estimatedTotalOperations()),
                         TransportRequestOptions.options().withTimeout(recoverySettings.internalActionTimeout()), EmptyTransportResponseHandler.INSTANCE_SAME).txGet();
             }
         });
 
         stopWatch.stop();
-        response.startTime = stopWatch.totalTime().millis();
-        logger.trace("{} recovery [phase2] to {}: start took [{}]",
+
+        response.startTime = stopWatch.totalTime().millis() - startEngineStart;
+        logger.trace("{} recovery [phase1] to {}: remote engine start took [{}]",
```

Comment:
Can you update the logging in `SharedFSRecoverySourceHandler`? The `phaseN` logging messages are now incorrect with the phase changes.